zebra-chain/src/amount.rs

```diff
@@ -205,6 +259,10 @@ pub enum Error {
         value: u64,
         source: std::num::TryFromIntError,
     },
+    /// overflow in multiplication
+    AmountMulError,
+    /// division error
+    AmountDivError(AmountDivError),
```

Comment:
Let's put the values in the errors?

And let's use a single error type, and have errors with specific names.

```suggestion
    /// i64 overflow when multiplying i64 non-negative amount {amount} by u64 {multiplier}
    MultiplicationOverflow {
        amount: i64,
        multiplier: u64,
    },
    /// division by zero is an invalid operation, amount {amount}
    DivideByZero { amount: i64 },
```